server/protocols/jmap/src/main/java/org/apache/james/jmap/JMAPServlet.java

```diff
@@ -18,92 +18,68 @@
+import com.fasterxml.jackson.databind.JsonNode;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.google.common.annotations.VisibleForTesting;
-import com.google.common.io.ByteStreams;
 import com.google.common.io.CharStreams;
 
 public class JMAPServlet extends HttpServlet {
 
-    public static class RawRequest {
-        public Object element;
-
-        public RawRequest(Object element) {
-            super();
-            this.element = element;
-        }
-
-        public RawRequest(String element) {
-            super();
-            this.element = element;
-        }
+    private static final long serialVersionUID = 1L;
 
-        public RawRequest(GetAccountsRequest element) {
-            super();
-            this.element = element;
-        }
-        
-        
-    }
-    
-    public static class Request {
-        public String action = "getAccounts";
-        public GetAccountsRequest arguments;
-        public String clientId;
-    }
-    
-    public static class GetAccountsRequest {
-        public String state;
-    }
-    
     public static final String JSON_CONTENT_TYPE = "application/json";
     public static final String JSON_CONTENT_TYPE_UTF8 = "application/json; charset=UTF-8";
 
-    private static final Logger LOG = LoggerFactory.getLogger(JMAPServlet.class);
-
     private ObjectMapper objectMapper = new ObjectMapper();
-    
-    private UsersRepository usersRepository;
+    private MethodDispatcher methodDispatcher;
 
     @Override
-    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
-        String username = "roger@barcamp";
-        String action = "getAccounts";
-        String input = CharStreams.toString(new InputStreamReader(req.getInputStream()));
-        
-        try (OutputStreamWriter output = new OutputStreamWriter(resp.getOutputStream())) {
-            output.append("[\"error\", {type: \"invalidArgument\"}, \"client-id\"]");
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException {
+        try {
+            String input = CharStreams.toString(new InputStreamReader(req.getInputStream()));
+            List<Protocol> requests = deserialize(input);
+            List<Protocol> responses = requests.stream()
+                .map(methodDispatcher::process)
+                .collect(Collectors.toList());
+
+            serialize(responses, resp.getOutputStream());
+        } catch (IOException e) {
+            resp.setStatus(SC_BAD_REQUEST);
         }
     }
-    
-    @VisibleForTesting RawRequest[] unserialize(String input) throws JsonParseException, JsonMappingException, IOException {
-        return objectMapper.readValue(input, RawRequest[].class);
+
+    @VisibleForTesting List<Protocol> deserialize(String input) throws IOException {
+        JsonNode[][] objects = objectMapper.readValue(input, JsonNode[][].class);
```

Comment:
ok, so nodes and objects ... looks like the more generic terms you could have found. Can't we have a clue about what is inside this JsonNode[][] ?